src/wp-admin/network/users.php

```diff
@@ -91,11 +91,21 @@
 								}
 
 								$userfunction = 'all_spam';
-								$blogs        = get_blogs_of_user( $user_id, true );
 
-								foreach ( (array) $blogs as $details ) {
-									if ( ! is_main_site( $details->userblog_id ) ) { // Main site is not a spam!
-										update_blog_status( $details->userblog_id, 'spam', '1' );
+								/**
+								 * Filters whether the blog status for all of a user's blogs should be updated.
+								 *
+								 * @since 6.9.0
+								 *
+								 * @param bool $update_blog_status Whether to update the blog status. Default false.
+								 */
+								if ( apply_filters( 'handle_network_allusers_update_blog_status', false, $user_id, $doaction ) ) {
```

Comment:
The filter name here seems like it could be refined.

Reply:
Like it is specifically whether a user being marked as spam should also result in their blogs being marked as spam, right? That should be made clear from the filter name. Right now the filter doesn't really say to me what it is doing.